### PR TITLE
Release Tiny Alpine 3.5 Docker Image

### PR DESCRIPTION
**Pull request by osterman**
## what
* Use a 2-stage build process to reduce the size of the docker image

## why
* `golang:1.7-alpine` is not that minimal when compared to `alpine:3.5`

## depends on
* https://github.com/cloudposse/build-harness/pull/31
* https://github.com/cloudposse/build-harness/pull/32

## related
* https://github.com/cloudposse/github-authorized-keys/issues/23

### CODE REVIEW

File: Dockerfile.build [hunk 1]
```diff
@@ -0,0 +1,78 @@
+FROM golang:1.7-alpine
```
Comment (darend) [inline Dockerfile.build]: If you switch to golang:1.8-alpine you will get some caching benefits as it uses the alpine-3.5 image

File: Dockerfile [hunk 2]
```diff
@@ -1,44 +1,8 @@
-FROM golang:1.7-alpine
+FROM alpine:3.5
 
-COPY ./ /go/src/github.com/cloudposse/github-authorized-keys
+COPY ./releases/github-authorized-keys_linux_amd64 /usr/bin/github-authorized-keys
```
Comment (darend) [inline Dockerfile]: Why not do the copy here https://github.com/cloudposse/github-authorized-keys/pull/24/files#diff-3254677a7917c6c01f55212f86c57fbfL77

Having a copy statement here will invalidate all subsequent cache layers

**Comment by osterman** [comment]
> Whats the new image size?

171mb -> 7mb

https://hub.docker.com/r/cloudposse/github-authorized-keys/tags/

**Comment by osterman** [comment]
@goruha seeing this in the output:
```
{"job":"syncUsers","level":"error","msg":"Connection to github.com failed","subsystem":"jobs","time":"2017-05-22T22:28:18Z"}
```

We're not catching this in CI. I tried running an older build with same credentials and it works. Something is wrong with this build.

**Comment by osterman** [comment]
>  I tried running an older build with same credentials and it works. Something is wrong with this build.

Never mind. The `ca-certificates` package is not installed by default so SSL certificate validation was failing. Wish our error messages were more expressive.

**Pull request merged**